mmcv/runner/hooks/momentum_updater.py

```diff
@@ -49,28 +52,33 @@ def _set_momentum(self, runner, momentum_groups):
                 elif 'betas' in param_group.keys():
                     param_group['betas'] = (mom, param_group['betas'][1])
 
-    def get_momentum(self, runner, base_momentum):
+    def get_momentum(self, runner, base_momentum) -> float:
         raise NotImplementedError
 
-    def get_regular_momentum(self, runner):
+    def get_regular_momentum(self, runner) -> Union[list, Dict[str, list]]:
         if isinstance(runner.optimizer, dict):
-            momentum_groups = {}
+            assert isinstance(self.base_momentum, dict)
+            momentum_groups: Dict[str, List[float]] = {}
             for k in runner.optimizer.keys():
-                _momentum_group = [
+                _momentum_group: List[float] = [
                     self.get_momentum(runner, _base_momentum)
                     for _base_momentum in self.base_momentum[k]
                 ]
                 momentum_groups.update({k: _momentum_group})
             return momentum_groups
         else:
+            assert isinstance(self.base_momentum, list)
             return [
                 self.get_momentum(runner, _base_momentum)
                 for _base_momentum in self.base_momentum
             ]
 
-    def get_warmup_momentum(self, cur_iters):
+    def get_warmup_momentum(
+            self,
+            cur_iters: int) -> Union[List[float], Dict[str, List[float]]]:
 
         def _get_warmup_momentum(cur_iters, regular_momentum):
+            warmup_momentum = None
```

Comment:
Why we need to set `warmup_momentum=None`?

Reply:
this is to resolve "possibly unbound" warning from Pylance. Other solution to this is to use `else` inplace of `elif self.warmup == 'exp'` since `self.warmup` has already been check to have those 3 possible values, how do you think?

Reply:
OK, I thought adding an extra else after `'exp'` to raise an error could be better. The original implementation is at risk of returning an undefined value.

Reply:
yepp, I already added this in the new commit